File: frontend/src/types/product.ts
export interface ProductCondition {
  min: number
  max: number
  marge_bas: number
  marge_haut: number
  temps_resistance_bas_min_h?: number
  temps_resistance_haut_min_h?: number
}

export interface ProductDetail {
  type_stockage_logistique: string
  ignore_environment: boolean
  temperature: ProductCondition
  humidite: ProductCondition
}

export interface ProductListItem {
  name: string
  type_stockage: string
}

File: backend/schemas/product.py
from __future__ import annotations

from typing import Optional
from pydantic import BaseModel, Field


class ProductCondition(BaseModel):
    min: float
    max: float
    marge_bas: float
    marge_haut: float
    temps_resistance_bas_min_h: Optional[float] = None
    temps_resistance_haut_min_h: Optional[float] = None


class ProductDetail(BaseModel):
    type_stockage_logistique: str
    ignore_environment: bool = False
    temperature: ProductCondition
    humidite: ProductCondition


class ProductListItem(BaseModel):
    name: str
    type_stockage: str
